app/services/mapping/goodsshipment/GoodsShipmentBuilder.scala

```diff
@@ -16,12 +16,41 @@
 
 package services.mapping.goodsshipment
 import forms.Choice
+import javax.inject.Inject
 import services.cache.ExportsCacheModel
 import services.mapping.goodsshipment.consignment.ConsignmentBuilder
 import services.mapping.governmentagencygoodsitem.GovernmentAgencyGoodsItemBuilder
 import uk.gov.hmrc.http.cache.client.CacheMap
+import wco.datamodel.wco.dec_dms._2.Declaration
 import wco.datamodel.wco.dec_dms._2.Declaration.GoodsShipment
 
+class GoodsShipmentBuilder @Inject()(
+  goodsShipmentNatureOfTransactionBuilder: GoodsShipmentNatureOfTransactionBuilder,
+  consigneeBuilder: ConsigneeBuilder
+) {
+  def buildThenAdd(exportsCacheModel: ExportsCacheModel, declaration: Declaration): Unit = {
+    val goodsShipment = new GoodsShipment()
+
+    goodsShipmentNatureOfTransactionBuilder.buildThenAdd(exportsCacheModel.natureOfTransaction, goodsShipment)
+    consigneeBuilder.buildThenAdd(exportsCacheModel.parties.consigneeDetails, goodsShipment)
+    ConsignmentBuilder.buildThenAdd(exportsCacheModel, goodsShipment)
```

Comment:
Can we change these to use the IOC style please so we're consistent, means we can test it all a lot easier

Reply:
I'm part way through this work so not changed all builders as yet.